concrete/js/build/core/app/heartbeat.js

```diff
@@ -0,0 +1,28 @@
+/* jshint unused:vars, undef:true, browser:true, jquery:true */
+/* global CCM_DISPATCHER_FILENAME */
+
+;(function(global, $) {
+'use strict';
+
+var lastUserInteraction = new Date();
+var lastHeartbeat = new Date();
+$(window).on('mousemove keydown keyup', function() {
+    lastUserInteraction = new Date();
+});
+setInterval(
```

Comment:
This is a fine solution, but you can probably avoid using a setInterval and just ping when those events happen.

Reply:
> This is a fine solution, but you can probably avoid using a setInterval and just ping when those events happen.

Done in 950d08ee318238770ce41b451aab51bce78ca8fb